File: java_solutions/1835.find-xor-sum-of-all-pairs-bitwise-and.java
/*
 * @lc app=leetcode id=1835 lang=java
 *
 * [1835] Find XOR Sum of All Pairs Bitwise AND
 */

class Solution {
    public int getXORSum(int[] arr1, int[] arr2) {
        return Arrays.stream(arr1).reduce(0, (a, b) -> a ^ b) & Arrays.stream(arr2).reduce(0, (a, b) -> a ^ b);
    }
}

File: kotlin_solutions_1/1835.find-xor-sum-of-all-pairs-bitwise-and.kt
/*
 * @lc app=leetcode id=1835 lang=kotlin
 *
 * [1835] Find XOR Sum of All Pairs Bitwise AND
 */

class Solution { fun getXORSum(arr1: IntArray, arr2: IntArray): Int = arr1.fold(0, Int::xor) and arr2.fold(0, Int::xor) }